### ISSUE TITLE
Do not allow to redeem gift cards when purchasing other gift cards

### ISSUE THREAD

**Issue by AndreiMaz** [nopSolutions/nopCommerce#7318]
nopCommerce version: 4.70.4

Do not allow to redeem gift cards when purchasing other gift cards. So when a customer will try to apply a gift card, we should validate whether it could be used or not. We already have the same logic applied to discounts (do not allow to purchase gift cards) - please check the "ValidateDiscountAsync" method of the "DiscountService":

```
if (await _productService.HasAnyGiftCardProductAsync(cartProductIds))
{...}
```

Let's implement the same logic in "GiftCardService.IsGiftCardValidAsync" and "ShoppingCartController.ApplyGiftCard" methods

**Comment by AndreiMaz** [comment]
moved to https://github.com/nopSolutions/nopCommerce/issues/6681

**Comment by prathamghosalkar** [comment]
assign me this task

**Comment by exileDev** [comment]
Closed #7318
